.github/workflows/release.yml

```diff
@@ -0,0 +1,116 @@
+name: Release
+
+on:
+  push:
+    tags:
+      - 'v*'
+
+jobs:
+  build_image:
+    runs-on: ubuntu-20.04
+    env:
+      GO_VERSION: 1.17
+    steps:
+    - name: Checkout Code
+      uses: actions/checkout@v3
+
+    - name: Install Golang
+      uses: actions/setup-go@v3
+      with:
+        go-version: ${{ env.GO_VERSION }}
+
+    - name: Build Image
+      run: |
+        TAG=$GITHUB_REF_NAME
+        docker build -t goharbor/harbor-acceld:$TAG -f script/release/Dockerfile .
+
+    - name: Test Image
+      # Enable tty for docker
+      shell: 'script -q -e -c "bash {0}"'
+      run: |
+        TAG=$GITHUB_REF_NAME
+        docker run -v $PWD/misc/config/config.yaml.nydus.tmpl:/etc/acceld-config.yaml -it -d --rm -p 2077:2077 goharbor/harbor-acceld:$TAG /etc/acceld-config.yaml
+        sleep 5
+        curl -f http://127.0.0.1:2077/api/v1/health
+
+    - name: Login Registry
+      uses: docker/login-action@v1
+      with:
+        username: ${{ secrets.DOCKER_HUB_USERNAME }}
+        password: ${{ secrets.DOCKER_HUB_PASSWORD }}
+
+    - name: Push Image
+      run: |
+        TAG=$GITHUB_REF_NAME
+        # Only push for non pull_request
+        if [ "$GITHUB_EVENT_NAME" != "pull_request" ]; then
+          docker push goharbor/harbor-acceld:$TAG
+        fi
```

Comment:
How about using https://github.com/docker/build-push-action ?

Reply:
Updated, but my concern is that we need to do a simple test before pushing the image, and this action doesn't seem to do that?

Reply:
Thanks, fixed.